packages/e2e-test-utils-playwright/src/editor/publish-post.ts

```diff
@@ -28,12 +28,11 @@ export async function publishPost( this: Editor ) {
 		'role=region[name="Editor publish"i] >> role=button[name="Publish"i]'
 	);
 
-	const urlString = await this.page
-		.getByRole( 'region', { name: 'Editor publish' } )
-		.getByRole( 'textbox', { name: 'address' } )
-		.inputValue();
-	const url = new URL( urlString );
-	const postId = url.searchParams.get( 'p' );
+	await this.page
+		.getByRole( 'button', { name: 'Dismiss this notice' } )
+		.filter( { hasText: 'published' } )
+		.waitFor();
+	const postId = new URL( this.page.url() ).searchParams.get( 'post' );
```

Comment:
> (…) if the query param isn't available, there's something wrong with the test, and it's better to fail.

That's why [I asked](https://github.com/WordPress/gutenberg/pull/56421#discussion_r1401871321) if there's a situation where the post id would not be available after saving/publishing. If not, we should probably throw from this util instead of returning `null`, IIUC.


> The post query arg doesn't change after a post is published or saved as a draft.

Right, the subsequent saves would already have the `post` param. The only potential flakey would come from the first save, then. I guess this is fine, though it would not be the first time I see racing issues caused by assuming one element's state off of another element's state – in this case, URL off of the post-publish notice.

Reply:
> I guess this is fine, though it would not be the first time I see racing issues caused by assuming one element's state off of another element's state

E2E tests have to make those assumptions, IMO.

> If not, we should probably throw from this util instead of returning null, IIUC.

Somethings like this:

```js
if ( postId !== 'string' ) {
  throw new Error( 'Unable to retrieve post ID from URL.' );
}

return parseInt( postId, 10 );
```

Reply:
> E2E tests have to make those assumptions, IMO.

Not if we can wait for stuff! 😄 (post ID in this case)

> Somethings like this?

I realized now that we could get the ID via the `getCurrentPostId()` which we don't need to wait for (stores the value before saving the post)

```js
const postId = await this.page.evaluate( () => {
	return window.wp.data.select( 'core/editor' ).getCurrentPostId();
} );

return postId;
```

Reply:
...but we can also wait for the `post` param, like this:

```js
const postId = await this.page.waitForFunction( () => {
	const url = new URL( window.location.href );
	if ( url.searchParams.has( 'post' ) ) {
		return url.searchParams.get( 'post' );
	}

	return false;
} );

return parseInt( postId, 10 );
```

Reply:
Thank you, @WunderBart 🙇 Do you think we should update the code or keep it as it is for now?

Reply:
If it ain't broke, don't fix it, right? 😄 No pressure here - let's just keep that in mind in case it starts making trouble.

Reply:
Sounds good!